cmd/kube-apiserver/app/server.go

```diff
@@ -346,12 +364,12 @@ func (s *APIServer) Run(_ []string) error {
 	if err != nil {
 		return err
 	}
-	storageVersions := make(map[string]string)
-	if s.StorageVersion == "" {
-		s.StorageVersion = g.Version
+
+	storageVersions := generateStorageVersionMap(s.StorageVersion, s.StorageVersions)
+	if _, found := storageVersions[g.Group]; !found {
+		storageVersions[g.Group] = g.Version
```

Comment:
Do we want to be nice here? Since we set a sane default, if this isn't set that means the user deliberately set it to something wrong, and I'm thinking maybe we should not silently fix that.